common/argparse.h

```diff
@@ -0,0 +1,893 @@
+    llvm::StringRef name;
+
+    // An optional version string that will be rendered as part of version meta
+    // action by the library. While this library does not impose a machine
+    // parsable structure, users will expect this to be extractable and parsable
+    // in practice.
+    //
+    // Subcommands with an empty version will inherit the first non-empty parent
+    // version.
+    //
+    // Whether a (possibly inherited) version string is non-empty determines
+    // whether this library provides the version-printing meta actions via a
+    // `--version` flag or, if there are subcommands, a `version` subcommand.
+    llvm::StringRef version = "";
+
+    // Optional build information to include when printing the version.
+    llvm::StringRef build_info = "";
+
+    // An optional long-form help string for the command.
+    //
+    // When accessing a command's dedicated help output, this will form the main
+    // prose output at the beginning of the help message. When listing
+    // subcommands, the subcommand's `help` string will be used to describe it
+    // in the list.
+    //
+    // The help meta actions are available regardless of whether this is
+    // provided in order to mechanically describe other aspects of the command.
+    //
+    // This field supports multiple lines and uses the text block handling
+    // described in the top-level `Args` comment.
+    llvm::StringRef help = "";
+
+    // An optional short help string for the command.
+    //
+    // This should be very short, at most one line and ideally fitting within a
+    // narrow column of a line. If left blank, the first paragraph of text (up
+    // to the first blank line) in the `help` string will be used.
+    llvm::StringRef help_short = "";
+
+    // An optional custom block of text to describe the usage of the command.
+    //
+    // The usage should just be a series of lines with possible invocations of
+    // the command summarized as briefly as possible. Ideally, each variation on
+    // a single line. The goal is to show the *structure* of different command
+    // invocations, not to be comprehensive.
+    //
+    // When omitted, the library will generate these based on the parsing logic.
+    // The generated usage is expected to be suitable for the vast majority of
+    // users, but can be customized in cases where necessary.
+    llvm::StringRef usage = "";
+
+    // An optional epilog multi-line block of text appended to the help display
+    // for this command. It is only used for this command's dedicated help, but
+    // can contain extra, custom guidance that is especially useful to have at
+    // the very end of the output.
+    llvm::StringRef help_epilog = "";
+  };
+
+  // Commands are classified based on the action they result in when run.
+  //
+  // Commands that require a subcommand have no action and are just a means to
+  // reach the subcommand actions. This also distinguishes meta actions from
+  // normal ones.
```

Comment:
```suggestion
  // reach the subcommand actions. Meta actions are also distinguished
  // normal ones.
```
"This" here has an ambiguous antecedent. To me, the original sentence reads as "Commands that require a subcommand have no action and are just a means to reach the subcommand actions. This means of reaching the subcommand actions also distinguishes meta actions from normal ones."

Reply:
Yeah, original was confusing. I've gone with an even sharper distinction.